lib/plist-service/plist-service.js

```diff
@@ -0,0 +1,53 @@
+import tls from 'tls';
+import B from 'bluebird';
+import PlistServiceEncoder from './transformer/plist-service-encoder';
+import PlistServiceDecoder from './transformer/plist-service-decoder';
+import LengthBasedSplitter from '../util/trasformer/length-based-splitter';
+
+class PlistService {
+  constructor (socketClient) {
+    this._socketClient = socketClient;
+    this._decoder = new PlistServiceDecoder();
+    this._splitter = new LengthBasedSplitter(false, 1000000, 0, 4, 4);
+    this._socketClient.pipe(this._splitter).pipe(this._decoder);
+
+    this._encoder = new PlistServiceEncoder();
+    this._encoder.pipe(this._socketClient);
+
+  }
+
+  async sendPlistAndReceive (json) {
+    this.sendPlist(json);
+    return await this.receivePlist();
+  }
+
+  sendPlist (json) {
+    this._encoder.write(json);
+  }
+
+  async receivePlist () {
+    return await new B((resolve) => this._decoder.once('data', (data) => resolve(data)));
```

Comment:
is the promise always resolved or there might be a timeout?

Reply:
That is actually a good point. I should also add this to the usbmux endpoints. What is the easiest way of setting that with bluebird?

Reply:
http://bluebirdjs.com/docs/api/timeout.html